fine grained rbac (not fully approved but almost complete with peer/tech preview) (#7962)
open-cluster-management/rhacm-docs: secure_cluster/fine_grain_rbac_ui.adoc

[#fine-grain-rbac-cli]
= Implementing fine-grained role-based access control with the console (Technology Preview)

> *Technology Preview:* {acm} supports fine-grained role-based access control (RBAC). As a cluster administrator, you can manage and control permissions with the `ClusterPermission` resource, which controls permissions at the namespace level on managed clusters, as well as at the cluster level. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster, or virtual machine.

Reviewer: ```suggestion
*Technology Preview:* {acm} supports fine-grained role-based access control (RBAC). As a cluster administrator, you can manage and control permissions with the `ClusterPermission` resource, which controls permissions at the namespace level on managed clusters, as well as at the cluster level. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster.
```

Reviewer: See explanation in CLI doc for this change.

> *Technology Preview:* {acm} supports fine-grained role-based access control (RBAC). As a cluster administrator, you can manage and control permissions with the `ClusterPermission` resource, which controls permissions at the namespace level on managed clusters, rather than at the cluster level. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster, or virtual machine.

Reviewer: ```suggestion
*Technology Preview:* {acm} supports fine-grained role-based access control (RBAC). As a cluster administrator, you can manage and control permissions with the `ClusterPermission` resource, which controls permissions at the namespace level on managed clusters, as well as at the cluster level. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster, or virtual machine.
```

Reviewer: Not sure if my proposed change is the best wording, but I don't think we should make it sound like ClusterPermission can ONLY do permissions at namespace level. It can also do so at cluster level. Admins will use a combination of both. However the feature itself is more about the fine grained part (namespace level), so it is right to focus on that part.

Learn how to set up for fine-grained role-based access control (RBAC) and the `ClusterPermission` resource from the console.

*Required access:* Cluster administrator

To learn about {ocp-short} default and virtualization roles and permissions, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/virtualization/about#authorization_virt-security-policies[Authorization] in the {ocp-short} documentation.

See xref:../secure_cluster/rbac_implement_rhacm.adoc#rhacm-rbac-implement[Implementing role-based access control] for more details about {acm-short} role-based access.

.Prerequisites

See the following requirements to begin using fine-grained role-based access control:

. Your `MultiClusterHub` custom resource `spec.overrides.components` field for `search` must `enabled` to retrieve a list of managed clusters namespaces that can represent virtual machines that are used for access control.
. You need virtual machines.

[#assign-fine-grain-rbac]
== Assigning fine-grained role-based access control in the console

> You can assign users to manage virtual machines with fine-grained role-based access control. Action are disabled in the console if the user-role access is not permitted. Slide the *YAML* option on to see the metadata that you enter populate the the YAML editor.

Reviewer: ```suggestion
You can assign users to manage virtual machines with fine-grained role-based access control. Action are disabled in the console if the user-role access is not permitted. Slide the *YAML* option on to see the data that you enter populate the the YAML editor.
```

Reviewer: I think using metadata here is confusing, because in the yaml itself there is a specific metadata section. However we are displaying the whole yaml and not just metadata section.

You can grant access to the following roles for {ocp-virt-short}, which are extensions of the default roles:

- `kubevirt.io:view`: only view resources
- `kubevirt.io:edit`: modify resources
- `kubevirt.io:admin`: view, modify, delete resources; grant permissions

> *Important:* As an administrator, you need to add either `Role bindings` or `Cluster role binding` for a valid `ClusterPermission` resource. You can also choose to add both. One cluster permission is used for each managed cluster.

Reviewer: ```suggestion
*Important:* As an administrator, you need to add either `Role bindings` or `Cluster role binding` for a valid `ClusterPermission` resource. You can also choose to add both.
```

Reviewer: I think we should remove this part because in 2.14 tech preview, they may need to create multiple ClusterPermissions for a single cluster depending on their needs. Only 1 ClusterRoleBinding can be in a ClusterPermission, so if they need more than that then they will have to create multiple ClusterPermissions for a single cluster.

. Navigate to the your `MultiClusterHub` custom resource to edit the resource and enable the feature.
.. From the `local-cluster` view, click *Operators* > *Installed Operators* > *Advanced Cluster Management for Kubernetes*. 
.. Click the *MultiClusterHub* tab to edit the resource. 

> .. Slide the *YAML* option on to see the metadata in the YAML editor.

Reviewer: ```suggestion
.. Slide the *YAML* option on to see the data in the YAML editor.
```

Reviewer: See previous comment for explanation about changing metadata to data. We don't need to use the word data specifically, I just don't think it should be metadata.

.. In your `MultiClusterHub` custom resource `spec.overrides.components` field, set `fine-grained-rbac-preview` to `true` to enable the feature. Change the `configOverrides` specification to `enabled: true` in the YAML editor and save your changes. See the following example with `fine-grained-rbac-preview` enabled:

+
[source,yaml]
----
 - configOverrides: {}
      enabled: true
      name: fine-grained-rbac-preview
----
. Label your `local-cluster` with `environment=virtualization`. 
.. From the `All Clusters` view, click **Infrastructure** > *Clusters* >
.. Find your `local-cluster` and click *Actions* to edit. 
.. Add the `environment=virtualization` label and save your changes. See the following example:

+
[source,bash]
----
environment=virtualization
----
. Change the `policy-virt-clusterroles` value for the `remediationAction` to `enforce`, which adds the `kubevirt` `clusterroles` to the hub cluster.
.. Click *Governance* > *Policies*.

> .. Find the `policy-virt-clusterroles` policy and click *Actions* to change the `remediationAction` value to `enforce`.

Reviewer: We should include the same tip as in the CLI doc of enabling the later remediationAction only because there are 2.

Author: Done

    resources: ["managedclusters"]
    verbs: ["list","get","watch"]
  - apiGroups: ["cluster.open-cluster-management.io"]
    resources: ["managedclusters"]
    verbs: ["get"]
    resourceNames: ["cluster01", "cluster02", "cluster03"] <2>
----
1. You can use a different name, but must continue to use that name during the process.
2. Add the names of the managed clusters that you want your User or Group to access.

. Create a `ClusterRoleBinding`. 
.. From the `local-cluster` view, click *User Management* > *RoleBindings* > *Create bindings*.
.. Choose `Cluster-wide role binding` for the _Binding type_.
.. Add the `RoleBinding` name that matches the name of the `ClusterRole`, which for this example is `acm-vm-rbac-required`.
.. Add the matching role name, which for this example is also `acm-vm-rbac-required`.
.. For the _Subject_, select User or Group, enter the User or Group name, and save your changes.
. Create a `ClusterPermission` resource to grant permissions at the namespace level.
.. Click *Access control* > *Create permission*. 
.. In the _Basic information_ window, add the cluster name and the User or Group that is granted permission.

> .. Choose the cluster or virtual machine with the specific namespace for that permission.

Reviewer: ```suggestion
.. Choose the cluster for that permission.
```

Reviewer: Not sure what was trying to be said here. On the basic information page, the only options are cluster and name (of the clusterpermission). There are no virtual machines at all in ClusterPermission

. Add the `Role bindings` information, which sets permissions at the namespace level. 

> .. Add namespaces in the cluster or virtual machine.

Reviewer: ```suggestion
.. Add virtual machine namespaces in the cluster.
```

Reviewer: There are no virtual machines at all in ClusterPermission, only namespaces.

.. Add Users or Groups.
.. Add roles, such as `kubevirt.io:view`, for fine-grained role-based access control. You can choose a combination of `RoleBindings`.
. Add the `Cluster role binding` with the same information to set permissions at the cluster level. 
. Review and click *Create permission* to create `ClusterPermission` resource as you see in the following example:

+
[source,bash]
----
apiVersion: rbac.open-cluster-management.io/v1alpha1
kind: ClusterPermission
metadata:
  name: <cluster01-prod-admin> <1>
  namespace: <cluster01> <2>
spec:
  roleBindings:
    - name: cluster01-prod-admin <3>
      namespace: prod <4>
      roleRef:
        name: kubevirt.io:admin
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
      subjects:
        - kind: User <5>
          apiGroup: rbac.authorization.k8s.io
          name: user1 <6>

----

> 1. Specify the cluster name for the permissions.

Reviewer: Name can be anything here. This is just the name of the ClusterPermission object, not cluster name.

Author: changed

2. Specify the managed cluster namespace for permissions.

> 3. Use the cluster name for `RoleBindings`, which assigns roles to Users or Groups.

Reviewer: Name can be anything here as well. This is just the name of the RoleBinding that gets created on the managed cluster. It does not need to be the cluster name.

4. Specify the namespace in the managed cluster to which the User or Group is granted access.
5. Choose User or choose Group.
6. Specify the User or Group name.
. Check for a `Ready` status in the console.
. You can click *Edit permission* to edit the `Role bindings` and `Cluster role binding`.
. *Optional:* Click *Export YAML* to use the resources for GitOps or in the terminal.
. You can delete the `ClusterPermissions` resource when you are ready.

. *Optional:* If `observability` is enabled, create an additional `RoleBinding` on the hub cluster so that users can view virtual machine details in Grafana.
.. From the `local-cluster` view, click *User Management* > *RoleBindings* > *RoleBindings*.
.. Choose `Namespace role binding` for the _Binding type_.
.. Add the `observability-grafana-access``RoleBinding` name.
.. Choose `view` for the _Role name_.

> .. For the _Subject_, select User or Group, enter the User or Group name, and save your changes.

Author: Ready for review

[#fine-grain-rbac-cli]
= Implementing fine-grained role-based access control with the console (Technology Preview)

> *Technology Preview:* {acm} supports fine-grained role-based access control (RBAC). As a cluster administrator, you can manage and control permissions with the `ClusterPermission` resource, which controls permissions at the namespace level on managed clusters, as well as at the cluster level. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster or virtual machine.

Reviewer: ```suggestion
*Technology Preview:* {acm} supports fine-grained role-based access control (RBAC). As a cluster administrator, you can manage and control permissions with the `ClusterPermission` resource, which controls permissions at the namespace level on managed clusters, as well as at the cluster level. Grant permissions to a virtual machine namespace within a cluster without granting permission to the entire managed cluster.
```

+
[source,yaml]
----
 - configOverrides: {}
      enabled: true
      name: fine-grained-rbac-preview
----
. Label your `local-cluster` with `environment=virtualization`. 
.. From the `All Clusters` view, click **Infrastructure** > *Clusters* >
.. Find your `local-cluster` and click *Actions* to edit. 
.. Add the `environment=virtualization` label and save your changes. See the following example:

+
[source,bash]
----
environment=virtualization
----
. Change the `policy-virt-clusterroles` value for the `remediationAction` to `enforce`, which adds the `kubevirt` `clusterroles` to the hub cluster.
.. Click *Governance* > *Policies*.
.. Find the `policy-virt-clusterroles` policy and click *Actions* to change the `remediationAction` value to `enforce`. *Important:* Two `remediationAction` specifications are in the policy, but you only need to change the later `remediationAction`. This does not apply to the first template in the YAML file. See the following sample:

+
[source,yaml]
----
  remediationAction: enforce
----

> .. Slide the *YAML* option on to see the metadata in the YAML editor and save your changes. See the following YAML sample:

Reviewer: ```suggestion
.. Slide the *YAML* option on to see the data in the YAML editor and save your changes. See the following YAML sample:
```

+

> ----
> remediationAction: enforce
> ----

Reviewer: ```suggestion
```

Reviewer: This looks like a duplicate to me, this is already shown in lines 63-65

. Create a `ClusterRole` resource.
.. From the `local-cluster` view, click *User Management* > *Roles* > *Create Role*.
.. Add the following `ClusterRole` resource information to the YAML editor:

+
[source,yaml]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: acm-vm-rbac-required <1>
rules:
  - apiGroups: ["clusterview.open-cluster-management.io"]
    resources: ["kubevirtprojects"]
    verbs: ["list"]
  - apiGroups: ["clusterview.open-cluster-management.io"]
    resources: ["managedclusters"]
    verbs: ["list","get","watch"]
  - apiGroups: ["cluster.open-cluster-management.io"]
    resources: ["managedclusters"]
    verbs: ["get"]

>     resourceNames: ["cluster01", "cluster02", "cluster03"] <2>

Reviewer: ```suggestion
    resourceNames: ["<cluster01>", "<cluster02>", "<cluster03>"] <2>
```

.. Add the matching role name, which for this example is also `acm-vm-rbac-required`.
.. For the _Subject_, select User or Group, enter the User or Group name, and save your changes.
. Create a `ClusterPermission` resource to grant permissions at the namespace level.
.. Click *Access control* > *Create permission*. 
.. In the _Basic information_ window, add the cluster name and the User or Group that is granted permission.
.. Choose the cluster for that permission.
. Add the `Role bindings` information, which sets permissions at the namespace level. 
.. Add virtual machine namespaces in the cluster.
.. Add Users or Groups.
.. Add roles, such as `kubevirt.io:view`, for fine-grained role-based access control. You can choose a combination of `RoleBindings`.
. Add the `Cluster role binding` with the same information to set permissions at the cluster level. 
. Review and click *Create permission* to create `ClusterPermission` resource as you see in the following example:

+
[source,bash]
----
apiVersion: rbac.open-cluster-management.io/v1alpha1
kind: ClusterPermission
metadata:
  name: <cluster01-prod-admin> <1>
  namespace: <cluster01> <2>
spec:
  roleBindings:

>     - name: cluster01-prod-admin <3>

Reviewer: ```suggestion
    - name: <cluster01-prod-admin> <3>
```

>       namespace: prod <4>

Reviewer: ```suggestion
      namespace: <prod> <4>
```

      roleRef:
        name: kubevirt.io:admin
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
      subjects:
        - kind: User <5>
          apiGroup: rbac.authorization.k8s.io

>           name: user1 <6>

Reviewer: ```suggestion
          name: <user1> <6>
```

----
<1> Specify a name for the permissions.

> <2> Specify a namespace for permissions.

Reviewer: ```suggestion
<2> Specify a cluster name for permissions.
```

Reviewer: Same as previous examples of this, I think it is more user friendly to call this a cluster name rather than namespace. It is both, but what they really want is the cluster name that this permission will get assigned to.

<3> Specify a name for `RoleBindings`, which assigns roles to Users or Groups.
<4> Specify the namespace in the managed cluster to which the User or Group is granted access.
<5> Choose `User` or choose `Group`.
<6> Specify the User or Group `name`.

. Check for a `Ready` status in the console.
. You can click *Edit permission* to edit the `Role bindings` and `Cluster role binding`.
. *Optional:* Click *Export YAML* to use the resources for GitOps or in the terminal.
. You can delete the `ClusterPermissions` resource when you are ready.

. *Optional:* If `observability` is enabled, create an additional `RoleBinding` on the hub cluster so that users can view virtual machine details in Grafana.
.. From the `local-cluster` view, click *User Management* > *RoleBindings* > *RoleBindings*.
.. Choose `Namespace role binding` for the _Binding type_.

> .. Add the `observability-grafana-access``RoleBinding` name.

Reviewer: The name "observability-grafana-access" can be anything, the one provided is just an example. FYI.

> [#fine-grain-rbac-cli]

Reviewer 2: Using the same anchor that is used in the previous file
```suggestion
[#fine-grain-rbac-ui]
```

Reviewer 2: @swopebe

Learn how to set up for fine-grained role-based access control (RBAC) and the `ClusterPermission` resource from the console.

*Required access:* Cluster administrator

To learn about {ocp-short} default and virtualization roles and permissions, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/virtualization/about#authorization_virt-security-policies[Authorization] in the {ocp-short} documentation.

See xref:../secure_cluster/rbac_implement_rhacm.adoc#rhacm-rbac-implement[Implementing role-based access control] for more details about {acm-short} role-based access.

.Prerequisites

See the following requirements to begin using fine-grained role-based access control:

. Your `MultiClusterHub` custom resource `spec.overrides.components` field for `search` must `enabled` to retrieve a list of managed clusters namespaces that can represent virtual machines that are used for access control.

> . You need virtual machines.

Reviewer 2: Similar to previous comment. Apply if the intention is to have an unordered list
```suggestion
- Your `MultiClusterHub` custom resource `spec.overrides.components` field for `search` must `enabled` to retrieve a list of managed clusters namespaces that can represent virtual machines that are used for access control.
- You need virtual machines.
```

> [#assign-fine-grain-rbac]

Reviewer 2: ```suggestion
[#assign-fine-grain-rbac-ui]
```

Reviewer 2: @swopebe

== Assigning fine-grained role-based access control in the console

> You can assign users to manage virtual machines with fine-grained role-based access control. Action are disabled in the console if the user-role access is not permitted. Slide the *YAML* option on to see the data that you enter populate the the YAML editor.

Reviewer 2: ```suggestion
You can assign users to manage virtual machines with fine-grained role-based access control. Actions are disabled in the console if the user-role access is not permitted. Slide the *YAML* option on to see the data that you enter populate in the YAML editor.
```

Reviewer 2: @swopebe

You can grant access to the following roles for {ocp-virt-short}, which are extensions of the default roles:

- `kubevirt.io:view`: only view resources
- `kubevirt.io:edit`: modify resources
- `kubevirt.io:admin`: view, modify, delete resources; grant permissions

> *Important:* As an administrator, you need to add either `Role bindings` or `Cluster role binding` for a valid `ClusterPermission` resource. You can also choose to add both.

Reviewer 2: I believe this is how the resource types are written
```suggestion
*Important:* As an administrator, you need to add either `RoleBindings` or `ClusterRoleBindings` for a valid `ClusterPermission` resource. You can also choose to add both.
```

Reviewer 2: @swopebe

Author: @dockerymick  you are right about the resource, but this part actually came from the UI. I watched the demos and looked at the build back then when I first started. I will go through this again before committing to see if that is why I chose this format when writing their choices and get back to you.

> . Navigate to the your `MultiClusterHub` custom resource to edit the resource and enable the feature.

Reviewer 2: ```suggestion
. Navigate to your `MultiClusterHub` custom resource to edit the resource and enable the feature.
```

Reviewer 2: @swopebe

> .. From the `local-cluster` view, click *Operators* > *Installed Operators* > *Advanced Cluster Management for Kubernetes*. 

Reviewer 2: in the past i recall italicizing the names of views, panels, or tabs. But based on the [highlighting style guide](https://www.ibm.com/docs/en/ibm-style?topic=format-highlighting#highlighting-by-element), this should be bold instead of monospace

```suggestion
.. From the *local-cluster* view, click *Operators* > *Installed Operators* > *Advanced Cluster Management for Kubernetes*. 
```

Reviewer 2: @swopebe

.. Click the *MultiClusterHub* tab to edit the resource. 
.. Slide the *YAML* option on to see the data in the YAML editor.
.. In your `MultiClusterHub` custom resource `spec.overrides.components` field, set `fine-grained-rbac-preview` to `true` to enable the feature. Change the `configOverrides` specification to `enabled: true` in the YAML editor and save your changes. See the following example with `fine-grained-rbac-preview` enabled:

+
[source,yaml]
----
 - configOverrides: {}
      enabled: true
      name: fine-grained-rbac-preview
----

> . Label your `local-cluster` with `environment=virtualization`. 

Reviewer 2: need line break and consider bolding
```suggestion

. Label your `local-cluster` with `environment=virtualization`. 
```

Reviewer 2: @swopebe

Author: Bold? @dockerymick we would not bold an annotation, if that is what you mean by bold. (Really bold should only be for definition lists, clickable items, but maybe I am missing what you mean here.)

Reviewer 2: you're right, i was thinking about local-cluster view like here : https://github.com/stolostron/rhacm-docs/commit/a748802666fdd4f404e160fad80afa6e4e4ce9aa

the link break is needed tho

Author: Ok, so I don't bold that but we can look at it again...maybe it should be bolded for the "view" but not here. I look at the view as a window type...then again, you can click on it. But not this one. This is the problem with having the view written like the code.

.. From the `All Clusters` view, click **Infrastructure** > *Clusters* >

> .. Find your `local-cluster` and click *Actions* to edit. 

Reviewer 2: consider bold based on highlighting guidance

Reviewer 2: @swopebe

Author: Bold is not recommended for highlighting, but I may be misunderstanding. I wonder if we are all not doing the same thing here.

`local-cluster` in the UI is (**unfortunately**) written like this (though that is changing soon) -- in the code all over we use` `local-cluster`` and same in the doc. If we were to follow "wintitle" dita tag--which is what we would use in DITA, it would actually be _italicized_ but we cannot really do that here because it is written like code.

```
<step>
  <cmd>Click <uicontrol>Configure</uicontrol>.</cmd>
  <stepresult>The <wintitle>Configuration Options</wintitle> window
  opens with your last set of selections highlighted.</stepresult>
</step>
```

Author: see above comment.

.. Add the `environment=virtualization` label and save your changes. See the following example:

+
[source,bash]
----
environment=virtualization
----

. Change the `policy-virt-clusterroles` value for the `remediationAction` to `enforce`, which adds the `kubevirt` `clusterroles` to the hub cluster.
.. Click *Governance* > *Policies*.
.. Find the `policy-virt-clusterroles` policy and click *Actions* to change the `remediationAction` value to `enforce`. *Important:* Two `remediationAction` specifications are in the policy, but you only need to change the later `remediationAction`. This does not apply to the first template in the YAML file. See the following sample:

+
[source,yaml]
----
  remediationAction: enforce
----

.. Slide the *YAML* option on to see the data in the YAML editor and save your changes. See the following YAML sample:

+

> [source,yaml]

Reviewer 2: missing sample

Reviewer 2: @swopebe

. Create a `ClusterRole` resource.
.. From the `local-cluster` view, click *User Management* > *Roles* > *Create Role*.
.. Add the following `ClusterRole` resource information to the YAML editor:

+
[source,yaml]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: acm-vm-rbac-required <1>
rules:
  - apiGroups: ["clusterview.open-cluster-management.io"]
    resources: ["kubevirtprojects"]
    verbs: ["list"]
  - apiGroups: ["clusterview.open-cluster-management.io"]
    resources: ["managedclusters"]
    verbs: ["list","get","watch"]
  - apiGroups: ["cluster.open-cluster-management.io"]
    resources: ["managedclusters"]
    verbs: ["get"]
    resourceNames: ["cluster01", "cluster02", "cluster03"] <2>
----

> 1. You can use a different name, but must continue to use that name during the process.
> 2. Add the names of the managed clusters that you want your User or Group to access.

Reviewer 2: ```suggestion
<1> You can use a different name, but must continue to use that name during the process.
<2> Add the names of the managed clusters that you want your User or Group to access.
```

Reviewer 2: @swopebe

> . Create a `ClusterRoleBinding`. 

Reviewer 2: ```suggestion
. Create a `ClusterRoleBinding` resource. 
```

Reviewer 2: @swopebe

.. From the `local-cluster` view, click *User Management* > *RoleBindings* > *Create bindings*.
.. Choose `Cluster-wide role binding` for the _Binding type_.
.. Add the `RoleBinding` name that matches the name of the `ClusterRole`, which for this example is `acm-vm-rbac-required`.
.. Add the matching role name, which for this example is also `acm-vm-rbac-required`.
.. For the _Subject_, select User or Group, enter the User or Group name, and save your changes.
. Create a `ClusterPermission` resource to grant permissions at the namespace level.
.. Click *Access control* > *Create permission*. 

> .. In the _Basic information_ window, add the cluster name and the User or Group that is granted permission.

Reviewer 2: Based on the style guide, window names should not be highlighted or use double-quotation marks

Reviewer 2: @swopebe

Author: hmm. I think there may be some misunderstanding/debate. For the project since the beginning we should have been following the "wintitle" concepts of windows, fields, panes.

This has always been based on the dita tag, which italicizes these components. We started the doc like this. These need to be italicized for now, but we can examine what new guidance is or what the tag will do when we change. All my windows have used the `wintitle` concept.

Author: For tagging: `wintitle` and `varname` are both tags from dita that _italicize_ and are used for windows, variables...

but given RH and IBM are not in DITA right now, hard to find that guidance, though the project was based on this guidance.

We are not italicizing our variables like we should be--over time we got away from it. Dita will likely force that. I

 think we need to look at if we are all doing the same thing for "windows". Fields used to be handled via `wintitle` but it looks like the guidance is now bold.

**This is where we started and what we brought with us:**

https://stolostron.github.io/rhacm-docs/ui_writing

.. Choose the cluster for that permission.

> . Add the `Role bindings` information, which sets permissions at the namespace level. 

Reviewer 2: ```suggestion
. Add the `RoleBindings` resource information, which sets permissions at the namespace level. 
```

Reviewer 2: @swopebe

.. Add virtual machine namespaces in the cluster.
.. Add Users or Groups.

> .. Add roles, such as `kubevirt.io:view`, for fine-grained role-based access control. You can choose a combination of `RoleBindings`.

Reviewer 2: ```suggestion
.. Add roles, such as `kubevirt.io:view`, for fine-grained role-based access control. You can choose a combination of `RoleBindings` resource.
```

Reviewer 2: @swopebe

> . Add the `Cluster role binding` with the same information to set permissions at the cluster level. 

Reviewer 2: ```suggestion
. Add the `ClusterRoleBinding` resource with the same information to set permissions at the cluster level. 
```

Reviewer 2: @swopebe

. Review and click *Create permission* to create `ClusterPermission` resource as you see in the following example:

+
[source,yaml]
----
apiVersion: rbac.open-cluster-management.io/v1alpha1
kind: ClusterPermission
metadata:
  name: <cluster01-prod-admin> <1>
  namespace: <cluster01> <2>
spec:
  roleBindings:
    - name: cluster01-prod-admin <3>
      namespace: prod <4>
      roleRef:
        name: kubevirt.io:admin
        apiGroup: rbac.authorization.k8s.io
        kind: ClusterRole
      subjects:
        - kind: User <5>
          apiGroup: rbac.authorization.k8s.io
          name: user1 <6>
----
<1> Specify a name for the permissions.
<2> Specify a namespace for permissions.
<3> Specify a name for `RoleBindings`, which assigns roles to Users or Groups.
<4> Specify the namespace in the managed cluster to which the User or Group is granted access.
<5> Choose `User` or choose `Group`.
<6> Specify the User or Group `name`.

. Check for a `Ready` status in the console.

> . You can click *Edit permission* to edit the `Role bindings` and `Cluster role binding`.

Reviewer 2: ```suggestion
. You can click *Edit permission* to edit the `RoleBindings` and `ClusterRoleBinding` resources.
```

Reviewer 2: @swopebe

Author: same here, I think I wrote this to match the UI. I went through the process carefully -- so we may need to keep this.

> . *Optional:* Click *Export YAML* to use the resources for GitOps or in the terminal.

Reviewer 2: resources for Red Hat OpenShift GitOps or the open project GitOps? If Red Hat OpenShift GitOps, reminder to use the conref for it

Reviewer 2: @swopebe

Author: Good observation. GitOps is the concept, then products make it their flavor. 
https://www.gitops.tech/#what-is-gitops

Much like DevOps.

We could change it to the brand, but then we would need to test it and in tech preview, not sure all that has been tested.

. You can delete the `ClusterPermissions` resource when you are ready.

> . *Optional:* If `observability` is enabled, create an additional `RoleBinding` on the hub cluster so that users can view virtual machine details in Grafana.

Reviewer 2: ```suggestion
. *Optional:* If `observability` is enabled, create an additional `RoleBinding` resource on the hub cluster so that users can view virtual machine details in Grafana.
```

.. From the `local-cluster` view, click *User Management* > *RoleBindings* > *RoleBindings*.
.. Choose `Namespace role binding` for the _Binding type_.

> .. Add the `observability-grafana-access``RoleBinding` name.

Reviewer 2: Based on @mshort55 comment:
```suggestion
.. Add the name for the `RoleBinding` resource.
```

Author: I am going to leave this as is for now bc we are meeting to decide whether to use the name or variable.

.. Choose `view` for the _Role name_.

> .. For the _Subject_, select User or Group, enter the User or Group name, and save your changes.

Reviewer 2: ```suggestion
.. For the _Subject_, select *User* or *Group*, enter the User or Group name, and save your changes.
```
